test/transaction/index.test.js

```diff
@@ -71,6 +71,49 @@ describe('transaction/index', function () {
     });
   });
 
+  describe('builder', function () {
+    describe('when there is no available authentication method', function () {
+      beforeEach(function () {
+        enrollment = benrollment({
+          availableMethods: ['sms'],
+          phoneNumber: '+1111111'
+        });
+      });
+
+      it('throws NoMethodAvailableError', function () {
+        expect(function () {
+          btransaction({
+            transactionToken,
+            enrollmentAttempt: null,
+            enrollments: [enrollment],
+            availableEnrollmentMethods: ['push', 'otp', 'sms'],
+            availableAuthenticationMethods: []
```

Comment:
I believe this test doesn't test the case you found in mfa-widget.

From what I see in `isAbleToStart...`, this will fail on the global `availableAuthenticationMethods` being an empty list, while the case might be that the only enrollment available has `sms` while availableAuthenticationMethods is e.g. `['push', 'otp']`.

I suggest you add that case to the test.

A related though: Is it reasonable that the mfa-api on start flow sends you an enrollment that only has a method that can't be used in the flow?